math/src/main/java/smile/math/Math.java

```diff
@@ -924,64 +924,68 @@ public static void permutate(Object[] x) {
     }
 
     /** Merges multiple vectors into one. */
-    public static int[] c(int[]... x) {
+    public static int[] c(int[]... list) {
         int n = 0;
-        for (int i = 0; i < x.length; i++) {
+        for (int[] x: list) {
             n += x.length;
         }
 
         int[] y = new int[n];
-        for (int i = 0, k = 0; i < x.length; i++) {
-            for (int xi : x[i]) {
-                y[k++] = xi;
+        int i = 0;
+        for (int[] x: list) {
+            for (int xi: x) {
+                y[i++] = xi;
             }
         }
         return y;
     }
 
     /** Merges multiple vectors into one. */
-    public static float[] c(float[]... x) {
+    public static float[] c(float[]... list) {
         int n = 0;
-        for (int i = 0; i < x.length; i++) {
+        for (float[] x: list) {
             n += x.length;
         }
 
         float[] y = new float[n];
-        for (int i = 0, k = 0; i < x.length; i++) {
-            for (float xi : x[i]) {
-                y[k++] = xi;
+        int i = 0;
+        for (float[] x: list) {
+            for (float xi: x) {
+                y[i++] = xi;
             }
         }
         return y;
     }
 
     /** Merges multiple vectors into one. */
-    public static double[] c(double[]... x) {
+    public static double[] c(double[]... list) {
         int n = 0;
-        for (int i = 0; i < x.length; i++) {
+        for (double[] x: list) {
             n += x.length;
         }
 
         double[] y = new double[n];
-        for (int i = 0, k = 0; i < x.length; i++) {
-            for (double xi : x[i]) {
-                y[k++] = xi;
+        int i = 0;
+        for (double[] x: list) {
+            for (double xi: x) {
+                y[i++] = xi;
             }
         }
         return y;
     }
 
-    /** Merges multiple vectors into one. */
-    public static String[] c(String[]... x) {
+    /** Concatenates multiple vectors into one array of strings. Uses .toString(), which is a no-op if already strings. */
```

Comment:
It is unlikely to merge multiple arrays to Strings. No need to make this function generic.